File: src/backend/Intelibill.Application/Features/Customers/DTOs/CustomerDto.cs
namespace Intelibill.Application.Features.Customers.DTOs;

public sealed record CustomerDto(
    Guid CustomerId,
    string Name,
    string PhoneNumber,
    string? Address,
    bool IsActive,
    decimal OutstandingDue,
    decimal CreditLimit,
    int PurchaseCount = 0,
    decimal LifetimeRevenue = 0m,
    decimal CurrentMonthRevenue = 0m);

File: src/mobile/android/intelibill_mobile/lib/src/features/customers/data/dto/customer_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'customer_dto.freezed.dart';
part 'customer_dto.g.dart';

@freezed
sealed class CustomerDto with _$CustomerDto {
  const factory CustomerDto({
    @JsonKey(name: 'customerId') required String customerId,
    @JsonKey(name: 'name') required String name,
    @JsonKey(name: 'phoneNumber') required String phoneNumber,
    @JsonKey(name: 'address') String? address,
    @JsonKey(name: 'isActive') required bool isActive,
    @JsonKey(name: 'outstandingDue') @Default(0.0) double outstandingDue,
  }) = _CustomerDto;

  factory CustomerDto.fromJson(Map<String, dynamic> json) =>
      _$CustomerDtoFromJson(json);
}
